crates/ruma-client-api/src/r0/media/get_content.rs

```diff
@@ -60,11 +60,9 @@ impl<'a> Request<'a> {
 
     /// Creates a new `Request` with the given url.
     pub fn from_url(url: &'a MxcUri) -> Result<Self, Error> {
-        if let Some((server_name, media_id)) = url.parts() {
-            Ok(Self { media_id, server_name, allow_remote: true })
-        } else {
-            Err(Error::InvalidMxcUri)
-        }
+        url.parts_err()
```

Comment:
This would be easier to read with `?` instead of `.map`. Same for the other changes in ruma-client-api.